### PR TITLE
localstore: lock all modified keys before commit

### PR DESCRIPTION
**Pull request by disksing**
for #782

### CODE REVIEW

File: store/localstore/txn.go [hunk 1]
```diff
@@ -35,17 +35,6 @@ type dbTxn struct {
 	snapshotVals map[string]struct{} // origin version in snapshot
```
Comment (shenli) [inline store/localstore/txn.go]: Pick a better name for this.
Maybe lockedKeys?

Comment (disksing) [inline store/localstore/txn.go]: Agreed : )

**Comment by shenli** [comment]
LGTM

File: store/localstore/txn.go [hunk 2]
```diff
@@ -127,6 +112,11 @@ func (txn *dbTxn) doCommit() error {
 
 	txn.ReleaseSnapshot()
 
+	txn.WalkBuffer(func(k kv.Key, v []byte) error {
+		txn.LockKeys(k)
```
Comment (ngaut) [inline store/localstore/txn.go]: Please return and check error.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**